### DISCUSSION TITLE
Prisma Detecting Drift for Foreign Keys

### DISCUSSION THREAD

**Discussion by rtnolan** [prisma/prisma#9188] [Q&A]
Hi there,

I have the following two models as part of my Prisma schema:

```prisma
model Position {
  portfolioId  String
  name         String
  group        String
  Transactions Transaction[]

  @@id([portfolioId, name, group])
}

model Transaction {
  id              Int      @id @default(autoincrement())
  portfolioId     String
  name            String
  group           String
  price           Float    @db.Float
  Position        Position @relation(fields: [portfolioId, name, group], references: [portfolioId, name, group])

  @@index([portfolioId, name, group], name: "txn_ibfk_1")
}
```

I'm using MySQL and viewing my Transaction table in MySQLWorkbench I can see that the table already has a foreign key named `txn_ibfk_1`. When I attempt to create a new migration for my database however I keep running into the error:
```
Drift detected: Your database schema is not in sync with your migration history.

The following is a summary of the differences between the expected database schema given your migrations files, and the actual schema of the database.

It should be understood as the set of changes to get from the expected schema to the actual schema.

[*] Changed the `Transaction` table
  [-] Removed foreign key on columns (portfolioId, name, group)
  [+] Added foreign key on columns (portfolioId, name, group)
 ```
  The migration which initially was applied to the database also has the following SQL in it:
  ```sql
  CREATE TABLE `Transaction` (
    `id` INTEGER NOT NULL AUTO_INCREMENT,
    `portfolioId` VARCHAR(191) NOT NULL,
    `name` VARCHAR(191) NOT NULL,
    `group` VARCHAR(191) NOT NULL,
    `price` FLOAT NOT NULL,
    `quantity` INTEGER NOT NULL,

    INDEX `txn_ibfk_1`(`portfolioId`, `name`, `group`),
    PRIMARY KEY (`id`)
) DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;
```
 I'm not quite sure what is causing Prisma to think there's a difference, any ideas?
 ```
 Prisma Version: 2.29.0
 OS: macOS Big Sur v11.5.1

**Comment by janpio** [comment]
This is most probably a bug. Can you help us out by providing all information required to reproduce the problem?

**Reply by ryands17** [reply]
@tnolan8 👋 

Yes it would be great if you could create an issue [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) with the reproduction steps 🙂

**Reply by rtnolan** [reply]
@ryands17  Can do. I believe I've got to the bottom of at least why it's happening. For the initial migration as I didn't want to reset the database (it was pre-existing) I created a separate MySQL database schema and generated the migration from that, I then applied the migration like: `prisma migrate resolve --applied init`. My foreign key on the Transaction table  has the name `txn_ibfk_1` and it appears Prisma is trying to rename it to `Transaction_ibfk_1`.

**Reply by rtnolan** [reply]
I believe all I need to do is upgrade to v3 and I can use the new Naming Constraints as explained [here](https://www.prisma.io/docs/guides/upgrade-guides/upgrading-versions/upgrading-to-prisma-3/named-constraints)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
